### PR TITLE
Fix document in NStepLSTM/GRU/RNN

### PR DESCRIPTION
**Pull request by aonotas**
I fixed typo in NStepLSTM/GRU/RNN functions.
I also fix document in NStepLSTM/RNN links to contain return variable in __call__.
This PR is related to #3903.

### CODE REVIEW

File: chainer/links/connection/n_step_lstm.py [hunk 1]
```diff
@@ -50,6 +50,22 @@ def __call__(self, hx, cx, xs, **kwargs):
...
+            ``hy``, ``cy`` and ``ys``.
+
+            - ``hy`` is an updated hidden states whose shape is the same as
+              ``hx``.
+            - ``cy`` is an updated cell states whose shape is the same as
+              ``cx``.
+            - ``ys`` is a list of :class:`~chainer.Variable` . Each element
+              ``ys[t]`` holds hidden states of the last layer corresponding
+              to an input ``xs[t]``. Its shape is ``(B_t, N)`` for
+              uni-directional LSTM and ``(B_t, 2N)`` for bi-directional LSTM
```
Comment (unnonouno) [inline chainer/links/connection/n_step_lstm.py]: ``B_t`` is the length of the t-th sequence, isn't it?

File: chainer/links/connection/n_step_rnn.py [hunk 2]
```diff
@@ -133,6 +133,19 @@ def __call__(self, hx, xs, **kwargs):
...
+        Returns:
+            tuple: This function returns a tuple containing three elements,
+            ``hy`` and ``ys``.
+
+            - ``hy`` is an updated hidden states whose shape is same as ``hx``.
+            - ``ys`` is a list of :class:`~chainer.Variable` . Each element
+              ``ys[t]`` holds hidden states of the last layer corresponding
+              to an input ``xs[t]``. Its shape is ``(B_t, N)`` for
+              uni-directional RNN and ``(B_t, 2N)`` for bi-directional RNN
+              where ``B_t`` is mini-batch size for time ``t``, and
```
Comment (unnonouno) [inline chainer/links/connection/n_step_rnn.py]: ditto

File: chainer/links/connection/n_step_lstm.py (hunk 1 above)
Comment (aonotas) [reply chainer/links/connection/n_step_lstm.py]: Thank you for your comment.
Yes, `B_t` is the length of t-th sequence.
I found that the document of `n_step_lstm` function contains also the wrong comment.
https://github.com/chainer/chainer/blob/v3.4.0/chainer/functions/connection/n_step_lstm.py#L296

File: chainer/functions/connection/n_step_gru.py [hunk 3]
```diff
@@ -93,9 +93,9 @@ def n_step_gru(
             hidden units.
         xs (list of chainer.Variable): A list of :class:`~chainer.Variable`
             holding input values. Each element ``xs[t]`` holds input value
-            for time ``t``. Its shape is ``(B_t, I)``, where ``B_t`` is
-            mini-batch size for time ``t``, and ``I`` is size of input units.
-            Note that this functions supports variable length sequences.
+            for time ``t``. Its shape is ``(L_t, I)``, where ``L_t`` is the
```
Comment (unnonouno) [inline chainer/functions/connection/n_step_gru.py]: I think the function node class gets transposed sequences. So, the original description is correct.

Comment (aonotas) [reply chainer/functions/connection/n_step_gru.py]: Oh, I see. Thank you for your comment!

**Comment by aonotas** [comment]
I fix documentation only Links. 
Please review it.

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**